big_tests/tests/mam_SUITE.erl

```diff
@@ -831,305 +696,157 @@ end_state(C, muc_light, Config) ->
-        true ->
-            Init()
+    skip_if(not lists:member(?config(configuration, Config), ConfList),
+            "message retraction not supported");
+maybe_skip(C, Config) when C =:= muc_light_failed_to_decode_message_in_database;
+                           C =:= pm_failed_to_decode_message_in_database ->
+    skip_if(?config(configuration, Config) =:= elasticsearch,
+            "elasticsearch does not support encodings");
+maybe_skip(C, Config) when C =:= muc_light_include_groupchat_filter;
+                           C =:= muc_light_no_pm_stored_include_groupchat_filter;
+                           C =:= muc_light_include_groupchat_messages_by_default ->
+    skip_if(?config(configuration, Config) =:= cassandra,
+            "include_groupchat field is not supported for cassandra backend");
+maybe_skip(C, Config) when C =:= easy_text_search_request;
+                           C =:= long_text_search_request;
+                           C =:= save_unicode_messages;
+                           C =:= muc_text_search_request ->
+    skip_if(?config(configuration, Config) =:= cassandra,
+            "full text search is not implemented for cassandra backend");
+maybe_skip(_C, _Config) ->
+    ok.
+
+skip_if(false, _Msg) -> ok;
+skip_if(true, Msg) -> {skip, Msg}.
+
+init_users(CaseName, Config) ->
+    case fresh_users(CaseName) of
+        [] ->
+            Config;
+        UserSpecs ->
+            escalus_fresh:create_users(Config, UserSpecs)
+    end.
+
+-define(requires_pm_archive(C),
+        C =:= querying_for_all_messages_with_jid;
+        C =:= query_messages_by_ids;
+        C =:= simple_query_messages_by_ids;
+        C =:= server_returns_item_not_found_for_ids_filter_with_nonexistent_id;
+        C =:= pagination_simple_enforced;
+        C =:= range_archive_request_not_empty;
+        C =:= limit_archive_request;
+        C =:= metadata_archive_request).
+
+-define(requires_muc_archive(C),
+        C =:= muc_query_messages_by_ids;
+        C =:= muc_simple_query_messages_by_ids;
+        C =:= muc_server_returns_item_not_found_for_ids_filter_with_nonexistent_id;
+        C =:= muc_querying_for_all_messages;
+        C =:= muc_querying_for_all_messages_with_jid;
+        C =:= muc_metadata_archive_request).
+
+fresh_users(C) when ?requires_pm_archive(C) ->
+    [{alice, 1}, {bob, 1}, {carol, 1}];
+fresh_users(C) when C =:= offline_message;
+                    C =:= archived;
+                    C =:= no_elements;
+                    C =:= only_stanzaid;
+                    C =:= same_stanza_id;
+                    C =:= metadata_archive_request_empty;
+                    C =:= metadata_archive_request_one_message;
+                    C =:= archive_chat_markers;
+                    C =:= dont_archive_chat_markers ->
+    [{alice, 1}, {bob, 1}];
+fresh_users(C) ->
+    case lists:member(C, all_cases_with_room()) of
+        true -> [{alice, 1}, {bob, 1}];
+        false -> []
+    end.
```

Comment:
Sometimes we have a list of cases and we check membership, some others we have a macro with a list of conditions, and in another one we have the list of conditions without the macro, directly coded into the clause. Can we have the clauses more consistent? Perhaps a map where the keys are the test cases and the values are configs as maps for the case? Maybe that is also repetitive, but `requires_pm_archive` preparing three users one of which is `carol` is not descriptive enough either 🤔

Just thinking out loud, this suite is really huge and complex, it won't be obvious how to clean it all up at once.

Reply:
I wanted to have it short and concise, and I tested different options, and this one was the shortest one.
Are the clauses really hard to understand for you? Maps with keys would be a lot of additional boilerplate code.
The macro `requires_pm_archive` prepares three users, because `bootstrap_archive` needs exactly these three users.

There is consistency:
- macro is used when the list of conditions repeats,
- there is only one list: `all_cases_with_room`, because this is the only case where I could use already existing groups of cases without creating new ones,
- in other cases, I explicitly listed the test cases.